rules/coding-style/src/ClassNameImport/ClassNameImportSkipper.php

```diff
@@ -35,4 +38,29 @@ public function shouldSkipNameForFullyQualifiedObjectType(
 
         return false;
     }
+
+    public function isShortNameInUseStatement(Name $name): bool
+    {
+        $longName = $name->toString();
+        if (Strings::contains($longName, '\\')) {
+            return false;
+        }
+
+        return $this->isFoundInUse($name);
+    }
+
+    private function isFoundInUse(Name $name): bool
+    {
+        $uses = $name->getAttribute(AttributeKey::USE_NODES);
+        foreach ($uses as $use) {
+            $useUses = $use->uses;
+            foreach ($useUses as $useUse) {
+                if ($useUse->name->getLast() === $name->getLast()) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
```

Comment:
This should be decoupled in own service that implements `ClassNameImportSkipVoterInterface`

Reply:
Ah, this might require more refactoring. Skip it for now